lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -54,27 +59,48 @@ public TaskServiceImpl(TaskanaEngine taskanaEngine, TaskMapper taskMapper,
         this.taskanaEngineImpl = (TaskanaEngineImpl) taskanaEngine;
         this.taskMapper = taskMapper;
         this.objectReferenceMapper = objectReferenceMapper;
+        this.workbasketService =  taskanaEngineImpl.getWorkbasketService();
     }
 
     @Override
-    public Task claim(String id, String userName) throws TaskNotFoundException {
+    public Task claim(String id)  throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
+        return claim(id, false);
+    }
+
+    @Override
+    public Task claim(String id, boolean forceClaim)  throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
+        String userName = CurrentUserContext.getUserid();
+        return claim(id, userName, forceClaim);
+    }
+
+    public Task claim(String id, String userName, boolean forceClaim) throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
```

Comment:
should be private and only get invoked by _this.claim()_
UserName is not necessary as parameter, because CurrentUser.getUserName() is static.